metapool/rescale_counts.py

```diff
@@ -35,25 +35,18 @@ def _apply_model(df_log10_cpm, linear_models):
     return converted_reads, failed_colnames
 
 
-def convert_read_count_to_cell_count(df, metadata_features):
+def convert_read_count_to_cell_count_per_g_input(df, metadata_features, prep_info_samples):
     lengths = metadata_features.loc[df.index, "total_length"]
+    sample_weights = prep_info_samples.loc[df.index, "calc_mass_sample_aliquot_input_g"]
     # Careful with types here.  If you use ints,
     # the length * 650 * 10**9 can overflow integers with very long genomes
-    mult_row = (6.022 * (10.0 ** 23)) / (lengths * (650 * 10.0 ** 9))
+    mult_row = (6.022 * (10.0 ** 23)) / (lengths * (650 * 10.0 ** 9) * sample_weights)
```

Comment:
@justinshaffer can you confirm that sample_weights will be less than 1? Based on the comment above the line, multiplying the line by another value larger than one will make it even easier for the denominator to overflow.

Reply:
Thanks @justinshaffer !